src/main/java/frc/team997/lib/util/DashboardChooser.java

```diff
@@ -52,46 +57,90 @@ public interface Option {
     }
 
     /**
-     * Constructs a DashboardChooser from an enum implementing {@link Option}
+     * Constructs a DashboardChooser from an enum implementing {@link Option}.
      *
      * @param enumClass The {@code <enum>.class}
      * @param defaultOption The default enum option
-     * @return A new DashboardChooser object
-     * @param <T> The enum's type
+     * @param logChanges Whether to log changes in the chosen value to an on-robot log.
+     * @return A new DashboardChooser object.
+     * @param <T> The enum's type.
      */
     @SuppressWarnings("unchecked")
     public static <T> DashboardChooser<T> fromEnum(
-            Class<? extends Option> enumClass, Option defaultOption) {
+            Class<? extends Option> enumClass, Option defaultOption, boolean logChanges) {
         Map<String, T> options = new HashMap<>();
         for (Option entry : enumClass.getEnumConstants()) {
             options.put(entry.getDisplayName(), (T) entry);
         }
         options.put(defaultOption.getDisplayName(), (T) defaultOption);
-        return new DashboardChooser<T>(options, defaultOption.getDisplayName());
+        return new DashboardChooser<T>(options, defaultOption.getDisplayName(), logChanges);
+    }
+
+    /**
+     * Constructs a DashboardChooser from an enum implementing {@link Option}, saving changes to an
+     * on-robot log.
+     *
+     * @param enumClass The {@code <enum>.class}
+     * @param defaultOption The default enum option
+     * @return A new DashboardChooser object.
+     * @param <T> The enum's type.
+     */
+    public static <T> DashboardChooser<T> fromEnum(
+            Class<? extends Option> enumClass, Option defaultOption) {
+        return fromEnum(enumClass, defaultOption, true);
     }
 
     /**
-     * Constructs a DashboardChooser from a {@code Map<String, T>}
+     * Constructs a DashboardChooser from a {@code Map<String, T>}.
      *
-     * @param options The option list, as a Map<title, value>
-     * @param defaultOption The title of the default option
+     * @param options The option list, as a Map<title, value>.
+     * @param defaultOption The title of the default option.
+     * @param logChanges Whether to log changes in the chosen value to an on-robot log.
+     * @throws InvalidParameterException If the default option is neither {@code null} or part of
+     *     the options.
      */
-    public DashboardChooser(Map<String, T> options, String defaultOption) {
-        // I watched this break the drivetrain - every shuffleboard option should
-        // have a default, but if not the default should be explicitly passed as null.
+    public DashboardChooser(Map<String, T> options, String defaultOption, boolean logChanges)
+            throws InvalidParameterException {
```

Comment:
this shouldn't throw InvalidParameterException since it's an unchecked exception and we don't want to have to try catch every use of DashboardChooser (and any potential exception will happen at robot initialization)